apps/src/studio/bigGameLogic.js

```diff
@@ -132,8 +132,12 @@ BigGameLogic.prototype.updateSpriteX_ = function (spriteIndex, updateFunction) {
   // Current behavior is that as soon as we go offscreen, we reset to the other
   // side. We could add a delay if we want.
   if (!this.onscreen(newCenterX)) {
-    // reset to other side
-    this.resetSprite_(sprite);
+    // reset to other side if it is visible
+    if (sprite.visible) this.resetSprite_(sprite);
+  } else if (!sprite.visible) {
```

Comment:
Sprites get hidden on reset now, so this catches the sprite as soon as it's moved far enough for onscreen? to be false and makes it visible again.